apis/Google.Cloud.Diagnostics.Common/Google.Cloud.Diagnostics.Common/RpcRetryConsumer.cs

```diff
@@ -0,0 +1,209 @@
+﻿// Copyright 2017 Google Inc. All Rights Reserved.
+// 
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// 
+//     http://www.apache.org/licenses/LICENSE-2.0
+// 
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+using Google.Api.Gax;
+using Grpc.Core;
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+
+
+namespace Google.Cloud.Diagnostics.Common
+{
+    /// <summary>
+    /// An <see cref="IConsumer{T}"/> that will allow for the option to retry RPC failures and buffer
+    /// the data while the retries occur.
+    /// </summary>
+    internal class RpcRetryConsumer<T> : IConsumer<T>
```

Comment:
Not in this PR but I have them in https://github.com/iantalarico/google-cloud-dotnet/tree/retry-test and will patch up and send a PR for them once this PR is in.